internal/active.go

```diff
@@ -39,47 +39,66 @@ type ScanResultDfActive struct {
 	Dest_Mac  string
 }
 
-func DefaultScan() {
+// DefaultScan example: you can set desiredCIDR to "" to use interface mask,
+// or "192.168.0.0/28" to request scanning that CIDR (must be inside interface network).
+func DefaultScan(networkInterface string, targetCIDR string, ICMPMode bool) {
+
+	if ICMPMode {
+		ICMPScan()
+	} else {
+		ArpScan(networkInterface, targetCIDR)
+	}
+
+}
+
+func ArpScan(networkInterface string, targetCIDR string) {
+	fmt.Println("Starting ARP scan...")
 	var wg sync.WaitGroup
-	// Get a list of all interfaces.
-	ifaces, err := net.Interfaces()
+
+	// Find all devices
+	devices, err := pcap.FindAllDevs()
 	if err != nil {
 		panic(err)
 	}
 
-	// Get a list of all devices
-	devices, err := pcap.FindAllDevs()
+	// if networkInterface == "any" {
+	// 	ifaces, err := net.Interfaces()
+	// 	if err != nil {
+	// 		panic(err)
+	// 	}
+	// 	for _, iface := range ifaces {
+	// 		wg.Add(1)
+	// 		go func(iface net.Interface) {
+	// 			defer wg.Done()
+	// 			if err := scan(&iface, &devices, targetCIDR); err != nil {
+	// 				log.Printf("interface %v: %v", iface.Name, err)
+	// 			}
+	// 		}(iface)
+	// 	}
+	// } else {
```

Comment:
Let's delete the code as we will still have commit history in case we need to go back